src/pages/dashboard/index.tsx

```diff
@@ -0,0 +1,81 @@
+                <div className="flex flex-col md:flex-row justify-center items-center space-y-4 md:space-y-0 md:space-x-1 m-4 w-[100%]">
+                    <input
+                        type="text"
+                        placeholder="Enter the URL"
+                        className="w-full md:w-[80%] bg-gray-200 md:rounded-l-2xl p-4 md:rounded-none"
+                        value={url}
+                        onChange={(e) => {
+                            getUrl(e.target.value);
+                        }}
+                    />
+                    <button
+                        className="w-full md:w-auto bg-gray-200 md:rounded-r-2xl px-4 md:px-8 py-4 hover:bg-gray-300 mt-2 md:mt-0 md:rounded-none"
+                        onClick={() => {
+                            const randomString = generateRandomString();
+                            setUrl(`https://rds.li/${randomString}`);
+                        }}
+                    >
+                        Generate
+                    </button>
+                </div>
+                <div className="flex flex-col md:flex-row justify-center items-center space-y-4 md:space-y-0 md:space-x-1 m-4 w-[98%]">
+                    <input
+                        type="text"
+                        placeholder="Copy the URL"
+                        className="w-full md:w-[80%] bg-gray-200 md:rounded-l-2xl p-4 md:rounded-none"
+                        disabled
+                        value={shortUrl}
+                    />
+                    <div className="flex flex-row justify-center items-center space-x-1">
+                        <button
+                            className="w-full md:w-auto bg-gray-200  px-4 md:px-8 py-3 hover:bg-gray-300 mt-2 md:mt-0 md:rounded-none"
+                            onClick={() => {
+                                const randomString = generateRandomString();
+                                setUrl(`https://rds.li/${randomString}`);
+                            }}
+                        >
+                            <ReloadIcon />
+                        </button>
+                        <button
+                            className="w-full md:w-auto bg-gray-200 px-4 md:px-8 py-3 hover:bg-gray-300 mt-2 md:mt-0 md:rounded-none"
+                            onClick={() => {
+                                setUrl('');
+                            }}
+                        >
+                            <AddIcon />
+                        </button>
+                        <button
+                            className="w-full md:w-auto bg-gray-200 md:rounded-r-2xl px-4 md:px-8 py-3 hover:bg-gray-300 mt-2 md:mt-0 md:rounded-none"
+                            onClick={() => {
+                                navigator.clipboard.writeText(shortUrl);
+                            }}
+                        >
+                            <CopyIcon />
+                        </button>
+                    </div>
```

Comment:
can we make input and button a reusable component?

Reply:
@j24m will handle this, had discussion